clients/src/main/java/org/apache/kafka/common/utils/Checksums.java

```diff
@@ -30,21 +36,74 @@
  */
 public final class Checksums {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(Checksums.class);
+    private static final MethodHandle BYTE_BUFFER_UPDATE;
+
+    static {
+        MethodHandle byteBufferUpdate = null;
+        if (Java.IS_JAVA9_COMPATIBLE) {
+            try {
+                byteBufferUpdate = MethodHandles.publicLookup().findVirtual(Checksum.class, "update", MethodType.methodType(void.class, ByteBuffer.class));
+            } catch (Throwable t) {
+                LOGGER.warn("Cannot uses java.util.zip.Checksum::update(ByteBuffer): direct ByteBuffers checksum won't use optimized platform support", t);
+            }
+        }
+        BYTE_BUFFER_UPDATE = byteBufferUpdate;
+    }
+
     private Checksums() {
     }
 
+    /**
+     * Uses {@link Checksum#update} on {@code buffer}'s content, without modifying its position and limit.<br>
+     * This is semantically equivalent to {@link #update(Checksum, ByteBuffer, int, int)} with {@code offset = 0}.
+     */
     public static void update(Checksum checksum, ByteBuffer buffer, int length) {
         update(checksum, buffer, 0, length);
     }
 
+    /**
+     * Uses {@link Checksum#update} on {@code buffer}'s content, starting from the given {@code offset}
+     * by the provided {@code length}, without modifying its position and limit.
+     */
     public static void update(Checksum checksum, ByteBuffer buffer, int offset, int length) {
         if (buffer.hasArray()) {
             checksum.update(buffer.array(), buffer.position() + buffer.arrayOffset() + offset, length);
-        } else {
-            int start = buffer.position() + offset;
-            for (int i = start; i < start + length; i++)
-                checksum.update(buffer.get(i));
+            return;
+        }
+        if (BYTE_BUFFER_UPDATE != null && buffer.isDirect()) {
```

Comment:
Do we have any tests for this method that use direct byte buffers? If not, can we please add them?

Reply:
Yep, already there